exporter/collector/monitoredresource.go

```diff
@@ -71,7 +75,15 @@ func resourceToLabels(
 		}
 		// Matches one of the resource filters
 		for _, resourceFilter := range resourceFilters {
-			if strings.HasPrefix(k, resourceFilter.Prefix) {
+			matchesRegex := true
+			if len(resourceFilter.Regex) > 0 {
+				matched, err := regexp.Match(resourceFilter.Regex, []byte(k))
+				if err != nil {
+					log.Debug("Unable to parse resource filter regex.", zap.Error(err), zap.Any("regex", resourceFilter.Regex))
+				}
+				matchesRegex = matched
+			}
+			if strings.HasPrefix(k, resourceFilter.Prefix) && matchesRegex {
```

Comment:
```suggestion
				if !matched {
				         continue
				}
			}
			if strings.HasPrefix(k, resourceFilter.Prefix) {
```
Seems simpler not to track matchesRegex separately

Reply:
Taking out the error handling made it easier for me to just do both checks in the `if`. I had trouble reusing the `matched` variable from the regexp call because I was declaring a new `err` on the same line. It's better now